apps/server/src/modules/authentication/services/ldap.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { System } from '@shared/domain';
+import { Client, createClient } from 'ldapjs';
+import { Logger } from '@src/core/logger';
+import { LdapConnectionError } from '../errors/ldap-connection.error';
+
+@Injectable()
+export class LdapService {
+	constructor(private logger: Logger) {
+		this.logger.setContext(LdapService.name);
+	}
+
+	async authenticate(system: System, username: string, password: string) {
+		const connection = await this.connect(system, username, password);
```

Comment:
I changed it to a `check...`-method. It does not return a boolean anymore. It throws if it fails otherwise it just finished with void like other check...-methods we have in the code base. That should make it more clear.